models/Asset/Thumbnail/ImageThumbnailTrait.php

```diff
@@ -336,6 +336,23 @@ protected function convertToWebPath(array $pathReference): ?string
         return $path;
     }
 
+    /**
+     * @return string
+     */
+    public function getFrontendPath(): string
+    {
+        $pathReference = $this->getPathReference();
+        $type = $pathReference['type'] ?? null;
+
+        if ($type === 'data-uri') {
+            return $pathReference['src'];
+        }
+
+        $path = $this->convertToWebPath($pathReference);
+
+        return Tool::getHostUrl() . $path;
+    }
```

Comment:
```suggestion
    /**
     * @return string
     */
    public function getFrontendPath(): string
    {
        $path = $this->getPath();
        if(!\preg_match('@^(https?|data):@', $path)) {
            $path = \Pimcore\Tool::getHostUrl() . $path;
        }

        return $path;
    }
```

This should do the trick, imho.